src/test_utils.rs

```diff
@@ -50,6 +54,13 @@ impl TestContext {
         let ctx = Context::new("FakeOS".into(), dbfile.into(), id)
             .await
             .unwrap();
+
+        let events = ctx.get_event_emitter();
+        async_std::task::spawn(async move {
+            while let Some(event) = events.recv().await {
+                receive_event(event.typ);
+            }
+        });
```

Comment:
That's not actually related to this PR but I needed the outputs (warnings etc.) during testing. It is in an extra commit so I could create an extra PR.